### DISCUSSION TITLE
Token injection when an API Gateway handles OIDC

### DISCUSSION THREAD

**Discussion by NMichas** [quarkusio/quarkus#26240] [Q&A] [answered]
Hello,

In my project I have an API Gateway which handles OIDC and then injects custom headers with the tokens before the request is proxied to my Quarkus services.

I was wondering, if instead of me writing a custom filter to extract the token from the headers and set the security principal there is a Quarkus way/filter to use instead? Quarkus has very rich OIDC integration and examples, however all samples I can find take for granted that Quarkus itself handles the authentication part.

Thanks!

[a comment by an automated account is not shown]

**Comment by NMichas** [comment, marked as answer]
https://quarkus.io/guides/security-jwt seems to do the job 👍

**Comment by sberyozkin** [comment]
@NMichas Sure, using smallrye-jwt custom factory should do
